File: design-hashmap.cc
const int MAX = 1e6;
int m[MAX + 1];

class MyHashMap {
public:
  MyHashMap() {
    memset(m, -1, sizeof(m));
  }
  
  void put(int key, int value) {
    m[key] = value;
  }
  
  int get(int key) {
    return m[key];
  }
  
  void remove(int key) {
    m[key] = -1;
  }
};

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */

File: d/design-hashmap.go
type MyHashMap struct {
	data []int
}

func Constructor() MyHashMap {
	slice := make([]int, 1000001)
	for i := range slice {
		slice[i] = -1
	}
	return MyHashMap{data: slice}
}

func (this *MyHashMap) Put(key int, value int) {
	this.data[key] = value
}

func (this *MyHashMap) Get(key int) int {
	return this.data[key]
}

func (this *MyHashMap) Remove(key int) {
	this.data[key] = -1
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Put(key,value);
 * param_2 := obj.Get(key);
 * obj.Remove(key);
 */
